### PR TITLE
Add test implementation for Language

### PR DESCRIPTION
**Pull request by madushajg**

### CODE REVIEW

File: components/cellery-component-parent.ipr [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nadundesilva) [inline components/cellery-component-parent.ipr]: Please remove this file from the commit. This is an Intellij IDEA project file.

File: components/cellery-component-parent.iws [hunk 2]
```diff
@@ -0,0 +1,418 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nadundesilva) [inline components/cellery-component-parent.iws]: Same as above. Please remove from commit.

File: components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (nadundesilva) [inline components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java]: Please update the year in the license header. Please fix in the other files as well

File: components/cellery-component-test/src/test/resources/testng.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+-->
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Ballerina-Kubernetes-Test-Suite">
+    <test name="ballerina-kubernetes-sample-tests" parallel="false">
+        <classes>
+            <!--<class name="org.cellery.components.test.HRTest"/>-->
```
Comment (nadundesilva) [inline components/cellery-component-test/src/test/resources/testng.xml]: Is disabling HRTest and EmployeeTest intentional ? Shouldn't we enable them ?

If so please add a TODO comment and a small explanation

File: components/cellery-component-test/src/test/java/org/cellery/components/test/utils/KubernetesTestUtils.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Kubernetes test utils.
+ */
+public class KubernetesTestUtils {
```
Comment (nadundesilva) [inline components/cellery-component-test/src/test/java/org/cellery/components/test/utils/KubernetesTestUtils.java]: The Class name should be updated to something more meaningful such as LangTestUtils.

File: testhw/helloworld/helloworld.bal [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import celleryio/cellery;
```
Comment (nadundesilva) [inline testhw/helloworld/helloworld.bal]: This seems to be an unused file. Please remove

File: components/cellery-component-test/pom.xml [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+        <artifactId>cellery-component-parent</artifactId>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>cellery-component-test</artifactId>
+    <packaging>jar</packaging>
+    <name>Cellery - Cellery Component Test</name>
+    <url>https://wso2-cellery.github.io</url>
+
+    <dependencies>
```
Comment (nadundesilva) [inline components/cellery-component-test/pom.xml]: Please verify whether you need all these dependencies. It looks like some of them are not required for this module.

File: components/pom.xml [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+        </repository>
+        <snapshotRepository>
+            <id>wso2.snapshots</id>
+            <name>WSO2 Snapshot Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/snapshots/</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+    <scm>
+        <url>https://github.com/ballerinax/kubernetes.git</url>
```
Comment (nadundesilva) [inline components/pom.xml]: Please update the SCM tag URLs properly

File: components/pom.xml [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+                <groupId>org.ballerinalang</groupId>
+                <artifactId>ballerina-packerina</artifactId>
+                <version>${ballerina.lang.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.ballerinax.docker</groupId>
+                <artifactId>docker-generator</artifactId>
+                <version>${docker.generator.version}</version>
+            </dependency>
+            <dependency>
```
Comment (nadundesilva) [inline components/pom.xml]: Do we need this dependency? Please remove all the unwanted dependencies

File: components/cellery-component-test/src/test/java/org/cellery/components/test/SampleTest.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.ballerinax.kubernetes.exceptions.KubernetesPluginException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public interface SampleTest {
```
Comment (nadundesilva) [inline components/cellery-component-test/src/test/java/org/cellery/components/test/SampleTest.java]: Do we need this interface? There is no benefit of having this.

File: components/lang/pom.xml [hunk 11]
```diff
@@ -201,7 +218,7 @@
                         </goals>
                         <configuration>
                             <createDependencyReducedPom>true</createDependencyReducedPom>
-                            <minimizeJar>true</minimizeJar>
+                            <!--<minimizeJar>true</minimizeJar>-->
```
Comment (nadundesilva) [inline components/lang/pom.xml]: Do we need to remove this ? If this temporary please add a TODO and a small explanation

File: components/cellery-component-test/pom.xml [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <goals>
+                    <goal>report</goal>
+                </goals>
+                <configuration>
+                    <dataFile>${project.build.directory}/coverage-reports/jacoco.exec</dataFile>
+                </configuration>
+            </plugin>
+            <plugin>
```
Comment (nadundesilva) [inline components/cellery-component-test/pom.xml]: Please move this and plugin and the same plugin in "lang" module to a pluginManagement section in the root POM

File: components/cellery-component-parent.ipr (hunk 1 above)
Comment (madushajg) [reply components/cellery-component-parent.ipr]: Removed

File: components/cellery-component-parent.iws (hunk 2 above)
Comment (madushajg) [reply components/cellery-component-parent.iws]: Removed

File: components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java (hunk 3 above)
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java]: Done

File: components/cellery-component-test/src/test/resources/testng.xml (hunk 4 above)
Comment (madushajg) [reply components/cellery-component-test/src/test/resources/testng.xml]: Yes. It is intentional.
Added in TODO

File: components/cellery-component-test/src/test/java/org/cellery/components/test/utils/CelleryUtils.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+        args = {@Argument(name = "imageName", type = TypeKind.STRING),
+                @Argument(name = "imageVersion", type = TypeKind.STRING),
+                @Argument(name = "cellImage", type = TypeKind.OBJECT),
+                @Argument(name = "instanceName", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.STRING)},
+        isPublic = true
+)
+
+public class CelleryUtils extends BlockingNativeCallableUnit {
+    public Cell getInstance(String destinationPath) {
```
[the remark this answers is not in the document]
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/utils/CelleryUtils.java]: Converted to static method

File: components/cellery-component-test/src/test/java/org/cellery/components/test/utils/KubernetesTestUtils.java (hunk 5 above)
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/utils/KubernetesTestUtils.java]: updated

File: testhw/helloworld/helloworld.bal (hunk 6 above)
Comment (madushajg) [reply testhw/helloworld/helloworld.bal]: Removed

File: components/pom.xml (hunk 9 above)
Comment (madushajg) [reply components/pom.xml]: Removed

File: components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+                "test-org", "test-img", "1.3.5"), 0);
+        File artifactYaml = TARGET_PATH.resolve("emp.yaml").toFile();
+        Assert.assertTrue(artifactYaml.exists());
+        CelleryUtils cellObj = new CelleryUtils();
+        cell = cellObj.getInstance(CELLERY_PATH.resolve("emp.yaml").toString());
+    }
+
+    @Test
+    public void validateCell() {
+        Assert.assertNotNull(cell);
```
[the remark this answers is not in the document]
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/EmployeeTest.java]: There can be. But for the moment the only check is availability.

File: components/cellery-component-test/src/test/java/org/cellery/components/test/utils/CelleryUtils.java [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+        functionName = "createInstance",
+        args = {@Argument(name = "imageName", type = TypeKind.STRING),
+                @Argument(name = "imageVersion", type = TypeKind.STRING),
+                @Argument(name = "cellImage", type = TypeKind.OBJECT),
+                @Argument(name = "instanceName", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.STRING)},
+        isPublic = true
+)
+
+public class CelleryUtils extends BlockingNativeCallableUnit {
```
[the remark this answers is not in the document]
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/utils/CelleryUtils.java]: Added private constructor.

Yes that shouldn't be a ballerina native function.

File: components/pom.xml (hunk 8 above)
Comment (madushajg) [reply components/pom.xml]: done

File: components/cellery-component-test/src/test/java/org/cellery/components/test/SampleTest.java (hunk 10 above)
Comment (madushajg) [reply components/cellery-component-test/src/test/java/org/cellery/components/test/SampleTest.java]: Dropped the interface

**Pull request closed without merging**